`google-cloud-cpp` build failure on x86_64-linux as of `891d0226`

## Issue description
Build of `google-cloud-cpp` failed on x86_64-linux as of 891d0226de6e281422a8d0d99814925fd7aed8d0. This is currently breaking python3Packages.optax.

```
error: builder for '/nix/store/q3ckz4zh2j25dxxhgzridvfdhvp690dg-google-cloud-cpp-1.38.0.drv' failed with exit code 1;
       last 10 log lines:
       > [852/2269] Linking CXX executable google/cloud/options_benchmark
       > FAILED: google/cloud/options_benchmark
       > : && /nix/store/58pwclg9yr437h0pfgrnbd0jis8fqasd-gcc-wrapper-11.2.0/bin/g++ -O3 -DNDEBUG  google/cloud/CMakeFiles/common_options_benchmark.dir/options_benchmark.cc.o -o google/cloud/options_benchmark  google/cloud/libgoogle_cloud_cpp_common.so.1.38.0  /nix/store/3lw7ca129dwhis7r0ygzkhqf1jqhsvv6-gbenchmark-1.6.1/lib/libbenchmark_main.a  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_bad_optional_access.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_time.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_civil_time.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_strings.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_int128.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_strings_internal.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_base.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_spinlock_wait.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_throw_delegate.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_time_zone.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_bad_variant_access.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_raw_logging_internal.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_log_severity.so.2103.0.1  /nix/store/3lw7ca129dwhis7r0ygzkhqf1jqhsvv6-gbenchmark-1.6.1/lib/libbenchmark.a  -lrt && :
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: google/cloud/libgoogle_cloud_cpp_common.so.1.38.0: undefined reference to `absl::lts_20210324::FormatTime[abi:cxx11](absl::lts_20210324::string_view, absl::lts_20210324::Time, absl::lts_20210324::TimeZone)'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: google/cloud/libgoogle_cloud_cpp_common.so.1.38.0: undefined reference to `absl::lts_20210324::string_view::find(absl::lts_20210324::string_view, unsigned long) const'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: google/cloud/libgoogle_cloud_cpp_common.so.1.38.0: undefined reference to `absl::lts_20210324::ByChar::Find(absl::lts_20210324::string_view, unsigned long) const'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: google/cloud/libgoogle_cloud_cpp_common.so.1.38.0: undefined reference to `absl::lts_20210324::strings_internal::CatPieces[abi:cxx11](std::initializer_list<absl::lts_20210324::string_view>)'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: google/cloud/libgoogle_cloud_cpp_common.so.1.38.0: undefined reference to `absl::lts_20210324::ParseTime(absl::lts_20210324::string_view, absl::lts_20210324::string_view, absl::lts_20210324::Time*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
       > collect2: error: ld returned 1 exit status
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/q3ckz4zh2j25dxxhgzridvfdhvp690dg-google-cloud-cpp-1.38.0.drv'.
error: 1 dependencies of derivation '/nix/store/qmnpxrpmh2a1bvd4sv9h3llslrz47wc6-arrow-cpp-7.0.0.drv' failed to build
building '/nix/store/h194vv9qddklp89ncdfisjcvpgj8jflq-mirrors-list.drv'...
building '/nix/store/ch6jgjnqviqvhysdx0dl924v75733qlj-python3.9-dm-tree-unstable-2021-12-20.drv'...
error: 1 dependencies of derivation '/nix/store/g5w9mpj5r4swalfcy9kdj3jqsfscjpm7-python3.9-pyarrow-7.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/p70222yd3mcjiqr7wys4csirpwhqky28-python3.9-apache-beam-2.37.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/a8m3088chlm9bs0bnzzmn8laf2al4n9a-python3.9-tensorflow-datasets-4.5.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xjgaf3h89jvpanlry1a57m2s2wl158iv-python3.9-optax-0.1.1.drv' failed to build
```

[full build log](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2181534947)

python3Packages.optax maintainers cc: @ndl
google-cloud-cpp maintainers cc: @cpcloud
Other cc: n/a

### Steps to reproduce
1. Checkout nixpkgs at commit 891d0226de6e281422a8d0d99814925fd7aed8d0
2. Run `nix-build -A google-cloud-cpp`

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.13.0-1021-azure, Ubuntu, 20.04.4 LTS (Focal Fossa), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.7.0`
 - nixpkgs: `/nix/store/r902mrj2iwsjhy8gs6y0rbr9nada0z4z-source`
```

## Misc.
This issue was automatically generated by [nixpkgs-upkeep](https://github.com/samuela/nixpkgs-upkeep).
- [CI workflow](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2181534947) that created this issue.
- Internal tag: 6261127071343039327

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -6431,7 +6431,11 @@ with pkgs;
 
   google-guest-oslogin = callPackage ../tools/virtualization/google-guest-oslogin { };
 
-  google-cloud-cpp = callPackage ../development/libraries/google-cloud-cpp { };
+  google-cloud-cpp = callPackage ../development/libraries/google-cloud-cpp {
+    abseil-cpp = abseil-cpp.override {
+      cxxStandard = "14";
+    };
+  };
 
   google-java-format = callPackage ../development/tools/google-java-format { };
 

